qiita_pet/handlers/study_handlers/artifact.py

```diff
@@ -21,3 +25,73 @@ def get(self):
         artifact = to_int(self.get_argument('artifact_id'))
         self.write(artifact_graph_get_req(artifact, direction,
                                           self.current_user.id))
+
+
+class ArtifactAJAX(BaseHandler):
+    def get(self):
+        artifact_id = to_int(self.get_argument('artifact_id'))
+        name = artifact_get_req(artifact_id)['name']
+        self.write(name)
+
+    def post(self):
+        artifact_id = to_int(self.get_argument('artifact_id'))
+        self.write(artifact_delete_req(artifact_id, self.current_user.id))
+
+
+class ArtifactAdminAJAX(BaseHandler):
+    def get(self):
+        artifact_id = to_int(self.get_argument('artifact_id'))
+        info = artifact_get_req(artifact_id, self.current_user.id)
+        status = info['visibility']
+        buttons = []
+
+        btn_base = ('<button onclick="set_admin_visibility(\'%s\', {0})" '
+                    'class="btn btn-primary">%s</button>').format(artifact_id)
+
+        if all([status == 'sandbox', qiita_config.require_approval]):
+            # The request approval button only appears if the processed data is
+            # sandboxed and the qiita_config specifies that the approval should
+            # be requested
+            buttons.append(
+                btn_base % ('awaiting_approval', 'Request approval'))
+        elif all([self.current_user.level == 'admin',
+                  status == 'awaiting_approval',
+                  qiita_config.require_approval]):
+            # The approve processed data button only appears if the user is an
+            # admin, the processed data is waiting to be approved and the qiita
+            # config requires processed data approval
+            buttons.append(btn_base % ('private', 'Approve artifact'))
+        elif status == 'private':
+            # The make public button only appears if the status is private
+            buttons.append(btn_base % ('public', 'Make public'))
+
+        # The revert to sandbox button only appears if the processed data is
+        # not sandboxed or public
+        if status not in {'sandbox', 'public'}:
+            buttons.append(btn_base % ('sandbox', 'Revert to sandbox'))
+
+        # Add EBI and VAMPS submission buttons if allowed
+        if all([not info['ebi_run_accessions'],
```

Comment:
Why are you using `all` and creating lists? Please, use the `and` keyword. Here and the next one.